rpc/serve.go

```diff
@@ -0,0 +1,69 @@
+package rpc
+
+import (
+	"context"
+	"errors"
+	"net"
+
+	"capnproto.org/go/capnp/v3"
+)
+
+// Serve serves a Cap'n Proto RPC to incoming connections.
+//
+// Serve will take ownership of bootstrapClient and release it after the listener closes.
+//
+// Serve exits with the listener error if the listener is closed by the owner.
+func Serve(lis net.Listener, bootstrapClient capnp.Client) error {
+	if !bootstrapClient.IsValid() {
+		err := errors.New("BootstrapClient is not valid")
+		return err
+	}
+	// Accept incoming connections
+	defer bootstrapClient.Release()
+	for {
+		conn, err := lis.Accept()
+		if err != nil {
+			// Since we took ownership of the bootstrap client, release it after we're done.
+			if !bootstrapClient.IsValid() {
```

Comment:
I don't think you need this check either.

Reply:
Looks like I forgot to release the bootstrap client here. Replacing the check with release.

Reply:
Hm? that's what the defer is for?

Reply:
Argh!